packages/react/menu/src/Menu.tsx

```diff
@@ -456,18 +461,43 @@ const MenuContentImpl = React.forwardRef((props, forwardedRef) => {
         <MenuContentProvider
           searchRef={searchRef}
           onItemEnter={(event) => {
-            if (isPointerInGraceArea(event, pointerGraceAreaRef.current)) event.preventDefault();
+            if (
+              isPointerMovingThroughGraceArea(
+                event,
+                pointerGraceAreaRef.current,
+                pointerXDirRef.current
+              )
+            )
+              event.preventDefault();
           }}
           onItemLeave={React.useCallback((event) => {
-            if (!isPointerInGraceArea(event, pointerGraceAreaRef.current)) {
+            const prematureLeave = pointerXDirRef.current !== pointerGraceAreaRef.current?.side;
+            if (
+              prematureLeave ||
```

Comment:
I'm struggling to see which case this covers?
If I'm leaving the item-trigger from the left side (in LTR) then I won't be in the grace area right? so the check is redundant.
I've removed the check locally and all seems to work still, but perhaps I'm missing the case this is catering for?

Reply:
Actually it seems to work because the same check already exists within `isPointerMovingThroughGraceArea` so regardless this one here isn't needed I think.

Reply:
I still think it's a bit odd reading this that this check would be needed at all. I'm guessing the reason is mostly because of the added bleed which will make the point actually still be inside the area?

Reply:
As discussed, this was redundant 😅